### ISSUE TITLE
Mapper conversions for Futures

### ISSUE THREAD

**Issue by sergeykolbasov** [finagle/finch#1013]
I had some thoughts on the easy migration to new `finchx` and one of them is to provide few additional implicit conversions for `Mapper` that will work with `Future` and automatically pack it into `E : Effect`.

So as an example:
```scala
import io.finch._
import io.finch.catsEffect._
import io.finch.futures._

val endpoint: Endpoint[IO, String] = get("hello") { Future.value(Ok("world"))) } //this apply will be automatically lifted into IO and converted to Mapper, so it should work
```

What do you think people? Is there any need for it?

**Comment by rpless** [comment]
I think there is definitely a need for this for both Twitter and Scala Futures (assuming we can support both). Changing to Cats Effect is a big change and not everyone is going to be able to move all of their codebases to it immediately (or even over the 6 months or whatever time frame that we're still going to publish Finch with futures). Easing the migration path will be beneficial for everyone.

**Comment by vkostyukov** [comment]
I like that. I wonder if we can do this w/o exporting `io.finch.futures._` and just provide mapping from both Twitter Future and Scala Future to whatever is the current effect type.

**Comment by sergeykolbasov** [comment]
No, I think `Mapper` is enough atm

**Comment by sergeykolbasov** [comment]
Closed with #1016
